src/java.base/share/classes/jdk/internal/foreign/Scoped.java

```diff
@@ -25,8 +25,12 @@
 
 package jdk.internal.foreign;
 
-import java.lang.foreign.ResourceScope;
+import java.lang.foreign.MemorySession;
 
 public interface Scoped {
-    ResourceScope scope();
+    MemorySessionImpl sessionImpl();
```

Comment:
This method is used to retrieve the "root" session associated with either a resource, or a session view. This means that, to simplify the implementation, `MemorySessionImpl` has to implement `Scoped` as well (and return itself).